### PR TITLE
chore: to add horizontal constraint to collapsible panel

### PR DESCRIPTION
**Pull request by YahiaElTai**
#### Summary

This PR adds a horizontal constraint to the collapsible panel with a default value of `scale`.
This change will not be a breaking change since the value is `scale` by default but will allow us to remove all usages of nested `Constraints.Horizontal` inside `CollapsiblePanel` in the MC.

**Note**: I have updated the `min-width` value on the `CollapsiblePanel` from `550px` to `284px` to match the new column system we have for the constraints but also to help with a use case we have in the merchant center where we need the panel to go lower than `550px`
The design team doesn't know why the value `550px` was assigned in the first place, and it seems like lowering it won't affect anything.

All changes in this PR have been approved by the design team.

### CODE REVIEW

**Comment by YahiaElTai** [comment]
I have updated percy snapshots, only one new prop added.

File: packages/components/collapsible-panel/src/collapsible-panel.styles.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ const getHeaderContainerStyles = (props, isOpen) => {
 
 const baseContainerStyles = css`
   position: relative;
-  min-width: 550px;
+  min-width: 284px;
```
Comment (jonnybel) [inline packages/components/collapsible-panel/src/collapsible-panel.styles.js:59]: I suppose this is intended to be the value of the smallest supported constraint

```suggestion
  min-width: ${vars.constraint6};
```

File: .changeset/chilled-chairs-ring.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"@commercetools-uikit/collapsible-panel": patch
```
Comment (jonnybel) [inline .changeset/chilled-chairs-ring.md:2]: I think we should consider these `feat`, since we're adding a new prop to the component 🙂

**Review by jonnybel — APPROVED** [review]
Looks good! Thanks :)

File: packages/components/collapsible-panel/package.json [hunk 3]
```diff
@@ -25,6 +25,7 @@
   "dependencies": {
     "@babel/runtime": "7.12.5",
     "@babel/runtime-corejs3": "7.12.5",
+    "@commercetools-uikit/constraints": "10.42.2",
```
Comment (emmenko) [inline packages/components/collapsible-panel/package.json:28]: Nit: can you sort the dep alphabetically?

File: packages/components/collapsible-panel/src/collapsible-panel.js [hunk 4]
```diff
@@ -211,6 +218,27 @@ CollapsiblePanel.propTypes = {
     // uncontrolled component does not have `onToggle` so no validation needed.
     return null;
   },
+  /**
+   * Horizontal size limit of the input fields.
+   */
+  horizontalConstraint: PropTypes.oneOf([
+    'm',
+    'l',
+    'xl',
```
Comment (emmenko) [inline packages/components/collapsible-panel/src/collapsible-panel.js:227]: I would leave them out, as this is a new prop and we don't need to be backwards compatible.

File: packages/components/collapsible-panel/src/collapsible-panel.styles.js [hunk 5]
```diff
@@ -56,7 +56,7 @@ const getHeaderContainerStyles = (props, isOpen) => {
 
 const baseContainerStyles = css`
   position: relative;
-  min-width: 550px;
+  min-width: ${vars.constraint6};
```
Comment (emmenko) [inline packages/components/collapsible-panel/src/collapsible-panel.styles.js:59]: 🎉

**Review by emmenko — CHANGES_REQUESTED** [review]
We should provide at least a couple of VRTs for the constraint sizes.

**Comment by emmenko** [comment]
The VRTs show some weird behavior:

Screenshot 2020-12-03 at 14 09 09

@YahiaElTai please don't approve the VRTs by yourself. One of the reviewer should do that. Thanks.

**Comment by YahiaElTai** [comment]
> The VRTs show some weird behavior:
> 
> Screenshot 2020-12-03 at 14 09 09
> 
> @YahiaElTai please don't approve the VRTs by yourself. One of the reviewer should do that. Thanks.

Ah so these snapshots are to be approved by the reviewers not me, sorry xD should have asked for help. I will take a look Thanks for the details

File: packages/components/collapsible-panel/src/collapsible-panel.js [hunk 6]
```diff
@@ -38,77 +39,83 @@ const CollapsiblePanel = (props) => {
...
-        <div
-          css={[baseContainerStyles, getThemeStyle(props.theme)]}
-          // Allow to override the styles by passing a `className` prop.
-          // Custom styles can also be passed using the `css` prop from emotion.
-          // https://emotion.sh/docs/css-prop#style-precedence
-          className={props.className}
+        <Constraints.Horizontal
+          max={Constraints.parseHorizontalConstraintProp(
+            props.horizontalConstraint
+          )}
```
Comment (emmenko) [inline packages/components/collapsible-panel/src/collapsible-panel.js:45]:
```suggestion
          max={props.horizontalConstraint}
```

The `parseHorizontalConstraintProp` is only used for backwards compatibility. We can simply require the new prop now as-is.

**Review by emmenko — COMMENTED** [review]
Thanks for the fix!

**Review by emmenko — APPROVED** [review]
Thanks! 🎉

**Comment by jonnybel** [comment]
The VRTs seemed to be good now 👍 The diffs are only due to the new prop and added specs.

**Comment by emmenko** [comment]
Let's go! Thanks again 👌

**Pull request merged**